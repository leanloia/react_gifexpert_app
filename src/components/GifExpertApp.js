import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Evangelion"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};

export default GifExpertApp;
