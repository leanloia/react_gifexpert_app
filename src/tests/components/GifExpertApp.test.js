import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";

describe("Pruebas del componente GifExpertApp", () => {
  test("debe mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["Evangelion", "Alison Brie"];

    // utilizamos 'defaultCategories' porque es la única forma de poder hacer un check
    // sobre un state particular (los state con hooks no tienen diferenciación entre si)
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
