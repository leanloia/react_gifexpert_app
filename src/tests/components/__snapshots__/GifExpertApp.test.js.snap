// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Evangelion"
      key="Evangelion"
    />
    <GifGrid
      category="Alison Brie"
      key="Alison Brie"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del componente GifExpertApp debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Peggy Olsen"
      key="Peggy Olsen"
    />
  </ol>
</Fragment>
`;
